### PR TITLE
recbole 1.2.0 :snowflake:

### PR DESCRIPTION
**Pull request by lorepirri**
recbole 1.2.0 :snowflake:

**Destination channel:** Snowflake

### Links

- [PKG-5807]
- dev_url (pypi): https://github.com/RUCAIBox/RecBole/tree/v1.2.0
- conda-forge: not there
- pypi: https://pypi.org/project/recbole/1.2.0
- pypi inspector: https://inspector.pypi.io/project/recbole/1.2.0/

### Depends on

- https://github.com/AnacondaRecipes/thop-feedstock/pull/1
- https://github.com/AnacondaRecipes/kmeans-pytorch-feedstock/pull/1
- https://github.com/AnacondaRecipes/setproctitle-feedstock/pull/1

### Explanation of changes:

- new feedstock
- additional missing dependency `kmeans-pytorch` not mentioned in the setup.py, open an issue about it here: https://github.com/RUCAIBox/RecBole/issues/2105
- add upstream tests.
- `colorlog` pinning relaxed to lower bound instead of exact. We have a more recent version and it seems to work (see screenshot).
- patch to update usage of `hyperopt` in a more recent version than specified in upstream (see patch comment).

image

[PKG-5807]: https://anaconda.atlassian.net/browse/PKG-5807?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ViridianMelody — APPROVED** [review]
LGTM pending a successful build upon removing the staging channels in abs.yaml.

File: recipe/meta.yaml
Comment (danpetry) [inline recipe/meta.yaml:22]: Do you want to build ray so you can get 3.12? I suppose this depends on whether snowflake wants to pay for that.

File: recipe/meta.yaml
Comment (danpetry) [inline recipe/meta.yaml:68]: should it be run_constrained? It's an extras_require

**Review by danpetry — APPROVED** [review]
Couple of minor comments but looks good

File: recipe/meta.yaml
Comment (lorepirri) [reply recipe/meta.yaml:22]: @danpetry  we are on hold for this one (see ticket). We do have ray in our pipeline, if it's out before we get the OK for this one then I will remove the skip

File: recipe/meta.yaml
Comment (lorepirri) [reply recipe/meta.yaml:68]: it is,

the suggested smoke test from upstream uses it, though (see my comment in the issue here: https://github.com/RUCAIBox/RecBole/issues/2105).

unfortunately, I did not receive any answer also for the rest of the seemingly optional dependencies (see comment a few lines before, and also in the above issue link).

Therefore, I moved all of those optional deps except some that we do not have to `run`.

Comment (lorepirri) [reply recipe/meta.yaml:68]: `recbole` upstream replied that the following dependencies are optional:

> `torch-scatter`, `dgl` and `kmeans-pytorch` are all optional.

ref: https://github.com/RUCAIBox/RecBole/issues/2105#issuecomment-2559262386

Therefore, I am moving the optional dependencies in `run_constrained`.

File: recipe/meta.yaml
Comment (lorepirri) [reply recipe/meta.yaml:22]: `ray-tune` does not have yet `py==3.12` variant.

```
ray-tune                       2.6.3 py310hca03da5_2  pkgs/main
ray-tune                       2.6.3 py311hca03da5_2  pkgs/main
ray-tune                       2.6.3  py38hca03da5_2  pkgs/main
ray-tune                       2.6.3  py39hca03da5_2  pkgs/main
```

skipping `py==312` because this ticket was on hold waiting for feedback for too long already.

**Pull request merged**
